### PR TITLE
chore(storybook): convert to csf v3 and enable lint

### PR DESCRIPTION
**Pull request by jinlee93**
### Summary:
- converts more stories into csf v3
- enables eslint rule for csf v3
[sc-187754]
[sc-193401]

### Test Plan:
- no test failures
- no relevant visual regression

### CODE REVIEW

**Comment by jinlee93** [comment]
Will follow up with converting more files to ts

**Comment by jinlee93** [comment]
Interestingly, chromatic is now capturing some pages, wonder why they weren't captured before

File: .eslintrc [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "plugin:react/recommended",
     "plugin:storybook/recommended",
     "plugin:testing-library/react",
-    "plugin:@chanzuckerberg/stories/recommended"
+    "plugin:@chanzuckerberg/stories/strict"
```
Comment (ahuth) [inline .eslintrc:22]: Let me know if you see any oddities with the no-csf-v2 and csf-v3-types rules.

**Review by ahuth — APPROVED** [review]
LGTM

File: .eslintrc (hunk 1 above)
Comment (jinlee93) [reply .eslintrc:22]: Sounds good, will do

**Pull request merged**